sled-agent/src/sled_agent.rs

```diff
@@ -338,6 +371,13 @@ impl SledAgent {
             }
         }
 
+        // Every sled runs an NTP zone, but in one of two flavours, depending
+        // on whether it is co-located with a switch zone. When co-located, the
+        // NTP zone runs in boundary mode and will synchronise time with
+        // servers outside of the rack as provided via RSS. Otherwise it runs
+        // as an internal server, sychronising with all boundary NTP servers.
+        self.configure_ntp(log, scrimlet).await;
```

Comment:
Just to confirm:

- If we boot a sled, but the Tofino driver is slow to load...
- ... so we `configure_ntp` with `scrimlet=false`
- ... but later on, the Tofino driver loads, and we call `self.configure_ntp(log, scrimlet = true)`...
- ... do we correctly remove the old NTP Zone, and load the Boundary NTP one?

Reply:
The way I've done this up to now (and I am happy to talk about alternatives) is that there is only one NTP zone image that is either acting as a boundary or not. To switch it from one to the other does not involve tearing it down and re-creating, it is just a matter of setting the SMF properties (`boundary` and `servers`), and refreshing the service. It reconfigures itself and restarts if necessary. You can test this in the NTP zone with:

```
svccfg -s ntp setprop config/boundary = false
svccfg -s ntp setprop config/server = test.example.com
svcadm refresh ntp
```

for example

```
root@oxz_ntp:~# sed -n 2p /etc/inet/chrony.conf
# Template configuration file for a boundary NTP server - one which

root@oxz_ntp:~# svccfg -s ntp setprop config/boundary = false
root@oxz_ntp:~# svcadm refresh ntp

root@oxz_ntp:~# sed -n 2p /etc/inet/chrony.conf
# Template configuration file for an internal NTP server - one which
```

Reply:
Excellent, that's exactly what I was wondering, and looks like it's working-as-intended.